samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/UpdateAssessmentQuestionsOrder.java

```diff
@@ -0,0 +1,172 @@
+		delegate = new ItemService();
+
+		boolean isEditPendingAssessmentFlow = author
+				.getIsEditPendingAssessmentFlow();
+
+		AssessmentBean assessmentBean = (AssessmentBean) ContextUtil.lookupBean("assessmentBean");
+
+
+		boolean published = isEditPendingAssessmentFlow ? false : true;
+
+		if(published) {
+			publisheddelegate = new PublishedAssessmentService();
+		} else {
+			assessdelegate = new AssessmentService();
+		}
+
+		reorderItems(published);
+
+		assessmentBean.setAssessment(published ? publisheddelegate.getAssessment(assessmentBean.getAssessmentId())
+				: assessdelegate.getAssessment(assessmentBean.getAssessmentId()));
+	}
+
+	public void reorderItems(boolean published) {
+		AssessmentBean assessmentBean = (AssessmentBean) ContextUtil
+				.lookupBean("assessmentBean");
+		
+		Map<Integer,Integer> swapSections = new HashMap<>();
+		Map<Integer,SectionFacade> sections = new HashMap<>();
+		
+		for(SectionContentsBean curSection : assessmentBean.getSections()) {
+			
+			SectionFacade sectFc = 	published ? 
+						(PublishedSectionFacade) publisheddelegate.getSection(curSection.getSectionId()) :
+						(SectionFacade) assessdelegate.getSection(curSection.getSectionId());
+			boolean dirty = false;
+			Integer sectionNumber = Integer.valueOf(curSection.getNumber());
+			Integer secVal = swapSections.get(sectionNumber);
+			if(sectionNumber != sectFc.getSequence()) {
+				SectionFacade sec = sections.get(sectionNumber);
+				if(sec != null) {
+					sec.setSequence(sectFc.getSequence());
+					swapSections.put(sec.getSequence(), sectFc.getSequence());
+					if(published) { 
+						publisheddelegate.saveOrUpdateSection(sec);
+					} else {
+						assessdelegate.saveOrUpdateSection(sec);
+					}
+				}
+				int val = sectionNumber;
+				swapSections.put(val,sectFc.getSequence());
+				sectFc.setSequence(val);
+				dirty = true;
+			} else if (secVal != null) {
+				// previous swap, change values
+				swapSections.put(sectFc.getSequence(),secVal);
+				sectFc.setSequence(secVal);
+				dirty = true;
+			}
+			if(dirty) {
+				if(published) { 
+					publisheddelegate.saveOrUpdateSection(sectFc);
+				} else {
+					assessdelegate.saveOrUpdateSection(sectFc);
+				}
+			}
+			sections.put(sectFc.getSequence(), sectFc);
+		}
+		sections.clear();
+		swapSections.clear();
+		updateItemsOrder(assessmentBean);
+	}
+	
+	private void updateItemsOrder(AssessmentBean assessmentbean) {
+		
+		for(SectionContentsBean curSection : assessmentbean.getSections()) {
+			
+			int numberOfItems = curSection.getQuestions();
+			
+			Map<Integer,Integer> swapItems = new HashMap<>();
+			Map<Integer,ItemFacade> sectionItems = new HashMap<>();
+			
+			for(ItemContentsBean icb : (List<ItemContentsBean>)curSection.getItemContents()) {
+				boolean dirty = false;
+				ItemData curItem = (ItemData) (icb).getItemData();
+				ItemFacade itemFc = delegate.getItem(String.valueOf(curItem.getItemId()));
+				Integer t = swapItems.get(curItem.getSequence());
```

Comment:
Sorry I just noticed this one too: there's a lot of calls to `curItem.getSequence()` which could be extracted to a variable. Not a huge deal if it's just a simple getter, but if there's any logic behind this method call it would be a good improvement.